src/types.rs

```diff
@@ -930,6 +932,14 @@ pub(crate) enum SigningAlgorithm {
     Hs256,
 }
 
+/// Attestation payload used for device-attest-01
```

Comment:
nit: I think it's worth a doc citation here too:

```suggestion
/// Attestation payload used for device-attest-01
///
/// See <https://datatracker.ietf.org/doc/draft-acme-device-attest/> for details.
```